fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/rdf/DefaultRdfStream.java

```diff
@@ -0,0 +1,321 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.rdf;
+
+import static java.util.Spliterator.IMMUTABLE;
+import static java.util.Spliterators.spliteratorUnknownSize;
+import static java.util.stream.Stream.empty;
+import static java.util.stream.StreamSupport.stream;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToIntFunction;
+import java.util.function.ToLongFunction;
+import java.util.stream.Collector;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+import com.hp.hpl.jena.graph.Node;
+import com.hp.hpl.jena.graph.Triple;
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.Statement;
+
+/**
+ * Implementation of a context-bearing RDF stream
+ *
+ * @author acoburn
+ * @since Dec 6, 2015
+ */
+public class DefaultRdfStream implements RdfStream {
+
+    protected Stream<Triple> stream;
+    protected final Node node;
+
+    /**
+     * Create an empty RdfStream
+     */
+    public DefaultRdfStream() {
+        this(empty());
+    }
+
+    /**
+     * Create an RdfStream
+     * @param node the topic of the stream
+     */
+    public DefaultRdfStream(final Node node) {
+        this(node, empty());
+    }
+
+    /**
+     * Create an RdfStream
+     *
+     * @param stream a stream of triples
+     */
+    public DefaultRdfStream(final Stream<Triple> stream) {
+        this.node = null;
+        this.stream = stream;
+    }
+
+    /**
+     * Create an RdfStream
+     * @param node the topic of the stream
+     * @param stream the incoming stream
+     */
+    public DefaultRdfStream(final Node node, final Stream<Triple> stream) {
+        this.node = node;
+        this.stream = stream;
+    }
+
+    /**
+     * Create an RdfStream from an existing Model.
+     * @param model An input Model
+     * @return a new RdfStream object
+     */
+    public static RdfStream fromModel(final Model model) {
+        return fromModel((Node)null, model);
```

Comment:
Do we even want to have a method that creates a stream with no topic?With hindsight, I would much rather we made the topic `final